apisix/core/etcd.lua

```diff
@@ -311,6 +311,9 @@ function _M.push(key, value, ttl)
     local index = res.body.header.revision
     index = string.format("%020d", index)
 
+    -- set the basic id attribute
+    value.id = index
```

Comment:
There is a problem, the consumer does not have an id. Adding an id directly here will cause the consumer data to become unusable.
@spacewander @okaybase

Reply:
thanks for your review firstly. 
Consumer regards the username property as the identity, so only the HTTP PUT method is supported for creating a new consumer.

This change is for creating a resource via POST. @nic-chen

Reply:
> thanks for your review firstly. 
> 
> Consumer regards the username property as the identity, so only the HTTP PUT method is supported for creating a new consumer.
> 
> 
> 
> This change is for creating a resource via POST. @nic-chen 

👍 thanks for the explanation.